### ISSUE TITLE
"function calls not allowed without using 'call'" error on attempt to print a field using direct access syntax

### ISSUE THREAD

**Issue by nd** [go-delve/delve#1736]
1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.3.2
Build: d064d1fe05b993e44a844a06df8c5c55c6c5ed5b
```
2. What version of Go are you using? (`go version`)?
```
go version go1.12.9 linux/amd64
```
3. What operating system and processor architecture are you using?
```
linux/amd64
```
4. What did you do?

Debugged the program with breakpoint on line 8:
```
package main

func main() {
	x := make(chan int, 10)
	x <- 1
	x <- 3
	x <- 5
	println(x) // break
}
```
and evaluated channels `buf` field using the direct access syntax `p *(*"[10]int")(824634261520)`.

5. What did you expect to see?
Contents of the buf field

6. What did you see instead?
Error `function calls not allowed without using 'call'`. Full delve log: https://gist.github.com/nd/6c1fe924252311cb859beddd6f8d3f0d

**Comment by derekparker** [comment]
Why not just `p x.buf`? 

```
Type 'help' for list of commands.
(dlv) b main.go:8
Breakpoint 1 set at 0x45925d for main.main() ./main.go:8
(dlv) c
> main.main() ./main.go:8 (hits goroutine(1):1 total:1) (PC: 0x45925d)
     3: func main() {
     4:         x := make(chan int, 10)
     5:         x <- 1
     6:         x <- 3
     7:         x <- 5
=>   8:         println(x) // break
     9: }
(dlv) p chan
Command failed: 1:5: expected type, found 'EOF'
(dlv) p x
chan int {
        qcount: 3,
        dataqsiz: 10,
        buf: *[10]int [1,3,5,0,0,0,0,0,0,0],
        elemsize: 8,
        closed: 0,
        elemtype: *runtime._type {
                size: 8,
                ptrdata: 0,
                hash: 4149441018,
                tflag: tflagUncommon|tflagExtraStar|tflagNamed (7),
                align: 8,
                fieldalign: 8,
                kind: 2,
                alg: *(*runtime.typeAlg)(0x4c70d0),
                gcdata: *1,
                str: 777,
                ptrToThis: 30720,},
        sendx: 3,
        recvx: 0,
        recvq: waitq<int> {
                first: *sudog<int> nil,
                last: *sudog<int> nil,},
        sendq: waitq<int> {
                first: *sudog<int> nil,
                last: *sudog<int> nil,},
        lock: runtime.mutex {key: 0},}
(dlv) p x.buf
*[10]int [1,3,5,0,0,0,0,0,0,0]
(dlv) 

```

**Comment by derekparker** [comment]
This does seem to be a valid bug however, the interpreter is incorrectly failing thinking your address cast is a function call. It is mentioned in the documentation and should be supported: https://github.com/go-delve/delve/blob/master/Documentation/cli/expr.md#nesting-limit.

**Comment by nd** [comment]
Direct access syntax is easier to use in client, this way to expand some nested child we don't have to construct an expression from a top level variable consisting of field/index/map key accesses.

**Comment by jrryjcksn** [comment]
This is not fixed in 1.5.0. I am currently seeing it with a newly built 1.5.0.

```
(dlv) p obj
k8s.io/apimachinery/pkg/apis/meta/v1/unstructured.Unstructured {
	Object: map[string]interface {} [
		"apiVersion": *(*interface {})(0xc00040fa68), 
		"kind": *(*interface {})(0xc00040fa78), 
		"metadata": *(*interface {})(0xc00040fa88), 
	],}
(dlv) p *((*"interface{}")(0xc00040fa78))
Command failed: function calls not allowed without using 'call'
```
